src/monitor/StakerMonitor.ts

```diff
@@ -0,0 +1,288 @@
+        transactionHash: typedEvent.transactionHash!,
+      });
+    }
+
+    for (const event of alteredEvents) {
+      const typedEvent = event as ethers.EventLog;
+      const { depositId, oldDelegatee, newDelegatee } = typedEvent.args;
+      await this.handleDelegateeAltered({
+        depositId,
+        oldDelegatee,
+        newDelegatee,
+        blockNumber: typedEvent.blockNumber!,
+        transactionHash: typedEvent.transactionHash!,
+      });
+    }
+  }
+
+  async handleStakeDeposited(event: StakeDepositedEvent): Promise<void> {
+    let attempts = 0;
+    while (attempts < this.config.maxRetries) {
+      const result = await this.eventProcessor.processStakeDeposited(event);
+      if (result.success || !result.retryable) {
+        return;
+      }
+      attempts++;
+      if (attempts < this.config.maxRetries) {
+        this.logger.warn(
+          `Retrying StakeDeposited event (attempt ${attempts + 1}/${this.config.maxRetries})`,
+          { event },
+        );
+        await new Promise((resolve) => setTimeout(resolve, 1000 * attempts)); // Exponential backoff
+      }
+    }
+    this.logger.error(
+      'Failed to process StakeDeposited event after max retries',
+      { event },
+    );
+  }
+
+  async handleStakeWithdrawn(event: StakeWithdrawnEvent): Promise<void> {
+    let attempts = 0;
+    while (attempts < this.config.maxRetries) {
+      const result = await this.eventProcessor.processStakeWithdrawn(event);
+      if (result.success || !result.retryable) {
+        return;
+      }
+      attempts++;
+      if (attempts < this.config.maxRetries) {
+        this.logger.warn(
+          `Retrying StakeWithdrawn event (attempt ${attempts + 1}/${this.config.maxRetries})`,
+          { event },
+        );
+        await new Promise((resolve) => setTimeout(resolve, 1000 * attempts));
+      }
+    }
+    this.logger.error(
+      'Failed to process StakeWithdrawn event after max retries',
+      { event },
+    );
+  }
+
+  async handleDelegateeAltered(event: DelegateeAlteredEvent): Promise<void> {
+    let attempts = 0;
+    while (attempts < this.config.maxRetries) {
+      const result = await this.eventProcessor.processDelegateeAltered(event);
+      if (result.success || !result.retryable) {
+        return;
+      }
+      attempts++;
+      if (attempts < this.config.maxRetries) {
+        this.logger.warn(
+          `Retrying DelegateeAltered event (attempt ${attempts + 1}/${this.config.maxRetries})`,
+          { event },
+        );
+        await new Promise((resolve) => setTimeout(resolve, 1000 * attempts));
+      }
+    }
+    this.logger.error(
+      'Failed to process DelegateeAltered event after max retries',
+      { event },
+    );
+  }
+
+  async getCurrentBlock(): Promise<number> {
+    return this.provider.getBlockNumber();
+  }
+
+  async getLastProcessedBlock(): Promise<number> {
+    return this.lastProcessedBlock;
+  }
+
+  async detectReorg(blockNumber: number): Promise<boolean> {
```

Comment:
Hmm can you help me understand what this function does and why we have it? Doesn't seem to be called internally.

Reply:
this isn't needed you're right